include/seqan/graph_algorithms/depth_first_search.h

```diff
@@ -121,28 +95,80 @@ void depthFirstSearch(TPredecessorMap & predecessor,
     typedef typename Iterator<TGraph, VertexIterator>::Type TVertexIterator;
     typedef typename VertexDescriptor<TGraph>::Type TVertexDescriptor;
     typedef typename Value<TPredecessorMap>::Type TPredVal;
+    typedef typename Iterator<Graph<TSpec>, AdjacencyIterator>::Type TAdjacencyIterator;
+
+    enum class Task : uint8_t
```

Comment:
Task is a little too general as a name. Since it resides in the global seqan namespace, could you give a little more specific name? If this is never used in a publicly documented interface, please also append an undercore to its name to mark it as "internal/private".